qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+            )
+
+    def _set_center(self, val: float) -> None:
+        """
+        Sets center frequency and updates start and stop frequencies if they
+        change.
+        """
+        self.write(f":SENSe:FREQuency:CENTer {val}")
+        self.update_trace()
+
+    def _set_span(self, val: float) -> None:
+        """
+        Sets frequency span and updates start and stop frequencies if they
+        change.
+        """
+        self.write(f":SENSe:FREQuency:SPAN {val}")
+        self.update_trace()
+
+    def _set_npts(self, val: int) -> None:
+        """
+        Sets number of points for sweep
+        """
+        self.write(f":SENSe:SWEep:POINts {val}")
+
+    def _enable_auto_sweep_time(self, val: str) -> None:
+        """
+        Enables auto sweep time
+        """
+        self.write(f":SENSe:SWEep:TIME:AUTO {val}")
+
+    def _enable_auto_sweep_type(self, val: str) -> None:
+        """
+        Enables auto sweep type
+        """
+        self.write(f":SENSe:SWEep:TYPE:AUTO {val}")
+
+    def _set_sweep_type(self, val: str) -> None:
+        """
+        Sets sweep type
+        """
+        self.write(f":SENSe:SWEep:TYPE {val}")
+
+    def _get_data(self, trace_num: int) -> ParamRawDataType:
+        """
+        Gets data from the measurement.
+        """
+        try:
+            timeout = self.sweep_time() + self.root_instrument._additional_wait
+            with self.root_instrument.timeout.set_to(timeout):
+                data_str = self.ask(f":READ:"
+                                    f"{self.root_instrument.measurement()}"
+                                    f"{trace_num}?")
+                data = np.array(data_str.rstrip().split(",")).astype("float64")
+        except TimeoutError as e:
+            raise RuntimeError("Couldn't receive any data. Command timed "
+                               "out.") from e
+
+        trace_data = data[1::2]
+        return trace_data
+
+    def update_trace(self) -> None:
+        """
+        Updates start and stop frequencies whenever span of/or center frequency
+        is updated.
+        """
+        self.start()
+        self.stop()
+
+    def setup_swept_sa_sweep(self,
+                             start: float,
+                             stop: float,
+                             npts: int) -> None:
+        """
+        Sets up the Swept SA measurement sweep for Spectrum Analyzer Mode.
+        """
+        self.root_instrument.mode("SA")
+        if "SAN" in self.root_instrument._available_meas():
+            self.root_instrument.measurement("SAN")
+        else:
+            raise RuntimeError("Swept SA measurement is not available on your "
+                               "Keysight N9030B instrument with Spectrum "
+                               "Analyzer mode.")
+        self.start(start)
+        self.stop(stop)
+        self.npts(npts)
+
+    def autotune(self) -> None:
+        """
+        Autotunes frequency
```

Comment:
I do not understand from this docstring what autotune does. Can we elaborate

Reply:
Added more information about it